### PR TITLE
python3Packages.flask-migrate: fix build

### LINKED ISSUE #138330: python39Packages.flask_migrate fails build/test: Error: No such template 'custom_template'
`python39Packages.flask_migrate` fails build as: https://hydra.nixos.org/build/153131050

```
test_migrate_upgrade (tests.test_custom_template.TestMigrate) ... 
$ flask db init -t ./custom_template

Error: No such template 'custom_template'

FAIL
```

### Notify maintainers

None.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: python39Packages.flask_migrate
```

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change
Test data was not part of the PyPi release.

Closes: #138330

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/flask-migrate/default.nix [hunk 1]
```diff
@@ -1,25 +1,47 @@
...
-  # tests invoke the flask cli which uses click and therefore has py3k encoding troubles
-  preCheck = lib.optionalString isPy3k ''
-    export LANG="en_US.UTF-8"
+  checkInputs = [
+    flask_script
+  ];
+
+  checkPhase = ''
+    runHook preCheck
+    ${python.interpreter} -m unittest discover
```
Comment (samuela) [inline pkgs/development/python-modules/flask-migrate/default.nix:37]: Is there a reason we can't use pytestCheckHook here?

File: pkgs/development/python-modules/flask-migrate/default.nix [hunk 2]
```diff
@@ -1,25 +1,47 @@
...
+    runHook postCheck
   '';
 
   meta = with lib; {
     description = "SQLAlchemy database migrations for Flask applications using Alembic";
-    license = licenses.mit;
     homepage = "https://github.com/miguelgrinberg/Flask-Migrate";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ];
```
Comment (samuela) [inline pkgs/development/python-modules/flask-migrate/default.nix:45]: Would you be willing to be a maintainer?

File: pkgs/development/python-modules/flask-migrate/default.nix [hunk 3]
```diff
@@ -1,25 +1,47 @@
...
+, flask
+, flask_script
+, flask_sqlalchemy
+, python
+}:
 
 buildPythonPackage rec {
   pname = "Flask-Migrate";
   version = "3.1.0";
+  format = "setuptools";
```
Comment (samuela) [inline pkgs/development/python-modules/flask-migrate/default.nix:15]: Isn't this already the default?

**Review by samuela — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/flask-migrate/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/flask-migrate/default.nix:37]: Yes, it uses pythons unittest module, not pytest.

**Comment by samuela** [comment]
Also please add a pythonImportsCheck

File: pkgs/development/python-modules/flask-migrate/default.nix (hunk 2 above)
Comment (mweinelt) [reply pkgs/development/python-modules/flask-migrate/default.nix:45]: No.

File: pkgs/development/python-modules/flask-migrate/default.nix (hunk 3 above)
Comment (mweinelt) [reply pkgs/development/python-modules/flask-migrate/default.nix:15]: We plan to migrate away from a default format in the future.

File: pkgs/development/python-modules/flask-migrate/default.nix (hunk 1 above)
Comment (samuela) [reply pkgs/development/python-modules/flask-migrate/default.nix:41]: Many packages can still be run through pytest even if it's not their official supported test runner. If pytest still doesn't work is there a reason that setuptoolsCheckHook doesn't work?

Comment (mweinelt) [reply pkgs/development/python-modules/flask-migrate/default.nix:41]: The `setuptoolsCheckHook` uses `python setup.py test` which is deprecated.

> WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point indep

We are mimicking the upstreams test environment. https://github.com/miguelgrinberg/Flask-Migrate/blob/943a938384deafcd1dedb10ead107827cf3324c9/tox.ini#L16

**Pull request merged**